server/task_executor.go

```diff
@@ -154,9 +156,42 @@ func executeMigration(ctx context.Context, server *Server, task *api.Task, state
 	if err != nil {
 		return 0, "", err
 	}
+
+	if task.Type == api.TaskDatabaseDataUpdate && task.Instance.Engine == db.MySQL {
```

Comment:
Shall we persist the thread ID before we migrate? This will guarantee the persistence of thread_id in case of any failure right after the migration.

Reply:
done.